### PR TITLE
evm-block-extractor readme and resume block extraction from latest block in zip file

### PR DESCRIPTION
**Pull request by veeso**
# Issue ticket

Issue ticket link: <>

### Code conventions

- [ ] I have performed a self-review of my code
- [ ] Every new function is documented
- [ ] Object names are auto explicative

### Security

- [ ] The PR does not break APIs backward compatibility
- [ ] The PR does not break the stable storage backward compatibility

### Testing

- [ ] Every function is properly unit tested
- [ ] I have added integration tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

File: src/evm-block-extractor/Cargo.toml [hunk 1]
```diff
@@ -16,8 +16,13 @@ env_logger = { workspace = true }
...
+ethereum-json-rpc-client = { path = "../ethereum-json-rpc-client", features = [
+  "reqwest",
+] }
 serde = { workspace = true }
 serde_json = { workspace = true }
 tokio = { workspace = true }
 zip = { workspace = true }
+
+[dev-dependencies]
+tempfile = "3"
```
Comment (ufoscout) [inline src/evm-block-extractor/Cargo.toml:28]: declare the dependency version inside the top Cargo.toml file

File: src/evm-block-extractor/src/blocks_reader.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+use std::path::Path;
+
+use zip::read::ZipArchive;
+
+pub struct BlocksReader {
+    reader: ZipArchive<File>,
+}
+
+impl BlocksReader {
+    pub fn open(path: &Path) -> anyhow::Result<Self> {
```
Comment (ufoscout) [inline src/evm-block-extractor/src/blocks_reader.rs:11]: `new`?

File: src/evm-block-extractor/src/blocks_reader.rs [hunk 3]
```diff
@@ -0,0 +1,71 @@
+use std::fs::File;
+use std::path::Path;
+
+use zip::read::ZipArchive;
+
+pub struct BlocksReader {
```
Comment (ufoscout) [inline src/evm-block-extractor/src/blocks_reader.rs:6]: Add a description

File: src/evm-block-extractor/src/blocks_reader.rs (hunk 2 above)
Comment (ufoscout) [inline src/evm-block-extractor/src/blocks_reader.rs:11]: Add a description

File: src/evm-block-extractor/src/blocks_reader.rs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+impl BlocksReader {
+    pub fn open(path: &Path) -> anyhow::Result<Self> {
+        let file = File::open(path)?;
+        let reader = ZipArchive::new(file)?;
+
+        Ok(Self { reader })
+    }
+
+    pub fn get_last_block_number(&mut self) -> anyhow::Result<u64> {
```
Comment (ufoscout) [inline src/evm-block-extractor/src/blocks_reader.rs:18]: Add a description

File: src/evm-block-extractor/src/blocks_reader.rs [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            anyhow::bail!("No blocks found");
+        }
+        let last_block_file = self.reader.by_index(self.reader.len() - 1)?;
+        let last_block_number = Self::block_number_from_file_name(last_block_file.name())?;
+
+        Ok(last_block_number)
+    }
+
+    fn block_number_from_file_name(file_name: &str) -> anyhow::Result<u64> {
+        let prefix = if file_name.starts_with("block") {
```
Comment (ufoscout) [inline src/evm-block-extractor/src/blocks_reader.rs:29]: if it does not start with "block" then it is not a block file so it should return an error

Comment (ufoscout) [inline src/evm-block-extractor/src/blocks_reader.rs:29]: Use constants for the PREFIX and SUFFIX and use them in both the writer and the reader

**Review by ufoscout — CHANGES_REQUESTED** [review]

**Review by itsyaasir — APPROVED** [review]

**Review by ufoscout — APPROVED** [review]

**Pull request merged**